wallet/tickets.go

```diff
@@ -439,3 +439,80 @@ func (w *Wallet) RevokeExpiredTickets(ctx context.Context, p Peer) (err error) {
 
 	return nil
 }
+
+// RevokeTicket creates and sends revocation transaction for the provided ticket
+// hash.  The wallet/account must be unlocked to generate any
+// revocations.
+func (w *Wallet) RevokeTicket(ctx context.Context, ticketHash *chainhash.Hash, p Peer) error {
+	const op errors.Op = "wallet.RevokeTicket"
+
+	feePerKb := w.RelayFee()
+	var revocation *wire.MsgTx
+	err := walletdb.View(ctx, w.db, func(dbtx walletdb.ReadTx) error {
+		addrmgrNs := dbtx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+		ticketPurchase, err := w.txStore.Tx(txmgrNs, ticketHash)
+		if err != nil {
+			return err
+		}
+
+		// Don't create revocations when this wallet doesn't have voting
+		// authority.
+		owned, havePrivKey, err := w.hasVotingAuthority(addrmgrNs, ticketPurchase)
+		if err != nil {
+			return err
+		}
+		if !owned || !havePrivKey {
+			return nil
```

Comment:
trying to revoke a ticket that we don't own, or have the key available for, should result in an error.